### PR TITLE
REF: Groupby._get_cythonized_result operate blockwise in axis==1 case

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -3010,15 +3010,49 @@ def blk_func(values: ArrayLike) -> ArrayLike:
...
+                )
+                if len(res_mgr.items) == 0:
+                    # We re-call grouped_reduce to get the right exception message
+                    try:
+                        mgr.grouped_reduce(blk_func, ignore_failures=False)
+                    except Exception as err:
+                        error_msg = str(err)
+                        raise TypeError(error_msg)
+                    # We should never get here
+                    raise TypeError("All columns were dropped in grouped_reduce")
```
Comment (jbrockmendel) [inline pandas/core/groupby/groupby.py:3047]: luckily we'll be able to get rid of L3028-L3047 once the deprecation is enforced

**Comment by jreback** [comment]
needs a rebase, b ut i think you have changed a bunch of this code

**Comment by jbrockmendel** [comment]
rebased + green

**Comment by jreback** [comment]
is this a bug fix? or just perf on a case?

**Comment by jbrockmendel** [comment]
> is this a bug fix? or just perf on a case?

Neither, just going from 2 paths to 1 path.

**Pull request merged**